### PR TITLE
BSDT fieldset changes

### PR DESCRIPTION
**Pull request by bmuniswamy**
# Critical Changes

# Changes
If the service field does not exist on the BSDT fieldsets then we will be automatically adding the field to the UI.
# Issues Closed

# New Metadata

# Deleted Metadata

# Definition of Done
  Refer to [Asteroids DoD document](https://salesforce.quip.com/iq2mAy4i62oM) to see any additional details for the items below
~~- [ ] Any net new LWC work has Sa11y tests & 50% or above lines JEST test coverage ~~
~~- [ ] CRUD/FLS is enforced in Apex~~
~~- [ ] Permission sets are updated to account for CRUD|FLS|Tab|Classes~~
~~- [ ] Field sets are updated to account for new fields~~
- [x] UX approval or UX not necessary
- [x] Link the pull request and work item by PR comment and Chatter post respectively, e.g. GUS: W-0000000: Work Name (https://github.com/SalesforceFoundation/PMM/pull/303)
- [ ] All **acceptance criteria** have been met
    - [x] Developer
    - [ ] Code Reviewer
    - [ ] QA
- [x] PR contains draft release notes
- [ ] QE story level testing completed

### CODE REVIEW

**Comment by bmuniswamy** [comment]
[W-9105446](https://gus.lightning.force.com/lightning/_classic/%2Fa07AH000000AFK4YAO)

File: force-app/main/default/classes/ServiceService.cls [hunk 1]
```diff
@@ -64,4 +67,42 @@ public with sharing class ServiceService {
...
+
+            for (Map<String, Object> field : fieldSetValues) {
+                String fieldApiName = (String) field.get('apiName');
+
+                if (fieldApiName == 'Contact__c') {
+                    indexToInsertServiceField = fieldSetValues.indexOf(field) + 1;
+                }
+                if (fieldApiName == 'ProgramEngagement__c') {
+                    indexToInsertServiceField = fieldSetValues.indexOf(field) + 1;
+                }
```
Comment (bethbrains) [inline force-app/main/default/classes/ServiceService.cls:92]: this is elegant!

**Review by bethbrains — CHANGES_REQUESTED** [review]
Looks good! Let's rename getFieldSets as discussed.

File: force-app/main/default/classes/ServiceService.cls (hunk 1 above)
Comment (AndersonTarren) [inline force-app/main/default/classes/ServiceService.cls:92]: Will these checks work for namespaced orgs?

Comment (bmuniswamy) [reply force-app/main/default/classes/ServiceService.cls:92]: I think in Apex they do

**Pull request merged**